examples/beacon-api-sidecar-fetcher/src/main.rs

```diff
@@ -0,0 +1,235 @@
+            match pending_result {
+                Ok(mined_sidecars) => {
+                    for sidecar in mined_sidecars {
+                        this.queued_actions.push_back(sidecar);
+                    }
+
+                    // Try returning the next available sidecar if any
+                    if let Some(sidecar) = this.queued_actions.pop_front() {
+                        return Poll::Ready(Some(Ok(sidecar)));
+                    }
+                }
+                Err(err) => return Poll::Ready(Some(Err(err))),
+            }
+        }
+
+        loop {
+            match this.events.poll_next_unpin(cx) {
+                Poll::Ready(Some(notification)) => {
+                    let notification_clone = notification.clone();
+                    let mut all_blobs_available = true;
+                    let mut actions_to_queue: Vec<MinedSidecar> = Vec::new();
+
+                    let txs: Vec<_> = notification
+                        .tip()
+                        .transactions()
+                        .filter(|tx| tx.is_eip4844())
+                        .map(|tx| {
+                            // Clone minimal data necessary for later processing
+                            (tx.hash, tx.blob_versioned_hashes().unwrap().len())
+                        })
+                        .collect();
+
+                    for (tx_hash, _) in &txs {
+                        match this.pool.get_blob(*tx_hash) {
+                            Ok(Some(blob)) => {
+                                actions_to_queue.push(MinedSidecar {
+                                    tx_hash: *tx_hash,
+                                    block_hash: notification.tip().block.hash(),
+                                    block_number: notification.tip().block.number,
+                                    sidecar: blob,
+                                });
+                            }
+                            Ok(None) => {
+                                all_blobs_available = false;
+                                break;
+                            }
+                            Err(err) => {
+                                return Poll::Ready(Some(Err(SideCarError::TransactionPoolError(
+                                    err,
+                                ))))
+                            }
+                        }
+                    }
+
+                    if all_blobs_available {
+                        this.queued_actions.extend(actions_to_queue);
+                    } else {
+                        let client_clone = this.client.clone();
+                        let block_hash = notification.tip().block.hash();
+                        let block_number = notification.tip().block.number;
+
+                        let url = format!(
+                            "http://{}/eth/v1/beacon/blob_sidecars/{}",
+                            "your-cl-node.com",
```

Comment:
the base url should be a field of the type

Reply:
Do you mean like the BeaconEventsConfig example?